packages/api/tests/lambdas/sf-scheduler.js

```diff
@@ -0,0 +1,126 @@
+'use strict';
+
+const rewire = require('rewire');
+const sinon = require('sinon');
+const test = require('ava');
+
+const { SQS } = require('@cumulus/ingest/aws');
+const schedule = rewire('../../lambdas/sf-scheduler');
+
+const queueName = 'userDefinedQueueName';
+const keyedItem = 'userDefinedQueueUrl';
+const defaultQueueName = 'startSF';
+const fakeMessageResponse = {
+  meta: {
+    queues: {
+      [queueName]: keyedItem,
+      [defaultQueueName]: 'startSFQueueUrl'
+    }
+  }
+};
+const scheduleEventTemplate = {
+  collection: 'fakeCollection',
+  provider: 'fakeProvider',
+  cumulusMeta: {},
+  payload: {},
+  template: 's3://somewhere/nice'
+};
+const fakeCollection = {
+  name: 'fakeCollection',
+  version: '000'
+};
+const fakeProvider = {
+  id: 'fakeProviderId',
+  host: 'fakeHost'
+};
+
+const buildMessage = schedule.__get__('buildMessage');
+const sqsStub = sinon.stub(SQS, 'sendMessage');
+
+// restore functions returned from rewire.__set__ commands to be called in afterEach
+let restoreList;
+
+test.afterEach(() => {
+  restoreList.map((restoreFn) => restoreFn());
+  sqsStub.resetHistory();
+});
+
+test.after.always(() => {
+  sqsStub.restore();
+});
+
+test.serial('Sends a message to SQS with queueName if queueName is defined', async (t) => {
+  restoreList = [
+    schedule.__set__('getMessageFromTemplate', () => Promise.resolve(fakeMessageResponse)),
+    schedule.__set__('getCollection', () => Promise.resolve(fakeCollection)),
+    schedule.__set__('getProvider', () => Promise.resolve(fakeProvider))
+  ];
+
+  const scheduleInput = { ...scheduleEventTemplate, queueName };
+  await schedule.schedule(scheduleInput);
+
+  t.is(sqsStub.calledOnce, true);
+
+  const [targetQueueUrl, targetMessage] = sqsStub.getCall(0).args;
+  t.is(targetQueueUrl, fakeMessageResponse.meta.queues[queueName]);
+  t.is(targetMessage.cumulus_meta.queueName, queueName);
+  t.deepEqual(targetMessage.meta.collection, fakeCollection);
+  t.deepEqual(targetMessage.meta.provider, fakeProvider);
+});
+
+test.serial('Sends a message to SQS with startSF if queueName is not defined', async (t) => {
+  restoreList = [
+    schedule.__set__('getMessageFromTemplate', () => Promise.resolve(fakeMessageResponse)),
+    schedule.__set__('getCollection', () => Promise.resolve(fakeCollection)),
```

Comment:
Why are we setting these in every test rather than the whole file if we're using them in every test?

Reply:
Just added a new commit. These can't all be defined at the top because at least one of the tests needs a different outcome from these functions.